modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/HashIndexDescriptor.java

```diff
@@ -89,26 +91,28 @@ public String toString() {
     /**
      * Creates an Index Descriptor from a given Table Configuration.
      *
-     * @param indexId Index ID.
      * @param tableConfig Table configuration.
+     *
      */
-    public HashIndexDescriptor(UUID indexId, TableView tableConfig) {
-        TableIndexView indexConfig = ConfigurationUtil.getByInternalId(tableConfig.indices(), indexId);
+    public HashIndexDescriptor(UUID indexId, TableView tableConfig, TablesConfiguration tablesConfig) {
```

Comment:
Here we mix `View` and `Configuration` classes, it looks bad.
You could pass view for index configuration here, right?